packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        return (
+            <Button
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                disabled={this.props.disabled}
+                onClick={this.handleToggleChange}
+                aria-label={this.renderARIALabel()}
+                appearance={
+                    ButtonAppearance[ActionToggleAppearance[this.getAppearance()]]
+                }
+                jssStyleSheet={actionToggleButtonOverrides}
+            >
+                {this.renderGlyph()}
+                {this.renderLabel()}
+            </Button>
+        );
+    }
+
+    public getAppearance(): ActionToggleAppearance {
+        if (this.state.selected) {
+            return ActionToggleAppearance.primary;
+        } else {
+            return ActionToggleAppearance.lightweight;
+        }
+    }
+
+    /**
+     * Returns the appropriate ARIA label
+     */
+    public renderARIALabel(): string {
+        if (this.state.selected) {
+            return this.props.selectedLabel;
+        } else {
+            return this.props.unselectedLabel;
+        }
+    }
+
+    /**
+     * Returns the appropriate text label
+     */
+    public renderLabel(): React.ReactNode {
+        if (this.state.selected) {
+            return this.props.selectedContent;
+        } else {
+            return this.props.unselectedContent;
+        }
+    }
+
+    /**
+     * Render Glyphs
+     */
+    public renderGlyph(): React.ReactNode {
+        if (this.state.selected) {
+            return this.renderSelectedGlyph();
+        } else {
+            return this.renderUnselectedGlyph();
+        }
+    }
+
+    public renderSelectedGlyph(): React.ReactNode {
+        if (typeof this.props.selectedGlyph === "function") {
+            return this.props.selectedGlyph(
+                get(this.props, "managedClasses.actionToggle_selectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    public renderUnselectedGlyph(): React.ReactNode {
+        if (typeof this.props.unselectedGlyph === "function") {
+            return this.props.unselectedGlyph(
+                get(this.props, "managedClasses.actionToggle_unselectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.actionToggle") || "";
+
+        if (this.props.disabled) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__disabled"
+            )}`;
+        }
+
+        if (this.state.selected) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__selected"
+            )}`;
+            classNames = `${classNames} ${this.getAppearanceString(
```

Comment:
Why do you need the `getAppearanceString` function at all? Why can't you just get the class name here?

Reply:
Actually don't need that function at all and can put the logic here.  I guess I thought I might need to use it elsewhere at some point.